sstables/sstable_directory.hh

```diff
@@ -206,6 +182,13 @@ public:
     // Retrieves the list of shared SSTables in this object. The list will be reset once this
     // is called.
     sstable_info_vector retrieve_shared_sstables();
+    std::vector<sstables::shared_sstable>& get_unshared_local_sstables() { return _unshared_local_sstables; }
+
+    future<> remove_input_sstables_from_resharding(std::vector<sstables::shared_sstable> sstlist);
+    future<> collect_output_sstables_from_resharding(std::vector<sstables::shared_sstable> resharded_sstables);
+
+    future<> remove_input_sstables_from_reshaping(std::vector<sstables::shared_sstable> sstlist);
+    future<> collect_output_sstables_from_reshaping(std::vector<sstables::shared_sstable> reshaped_sstables);
 
```

Comment:
I like very much we're removing reshape and reshard concept from sstable_directory, so it would be great to rename these functions accordingly.

It seems we don't need both collect_output_sstables_from_resharding() and collect_output_sstables_from_reshaping(). We can keep only the former (which serves well for both resharding and reshape) and rename it to collect_output_unshared_sstables().

remove_input_sstables_from_resharding() can become remove_input_sstables() as it does nothing special on the provided set, just remove the sstables in it.

remove_input_sstables_from_reshaping() is only for unshared, and can be renamed to remove_input_unshared_sstables().

Reply:
> I like very much we're removing reshape and reshard concept from sstable_directory, so it would be great to rename these functions accordingly.

Absolutely!

> It seems we don't need both collect_output_sstables_from_resharding() and collect_output_sstables_from_reshaping(). We can keep only the former (which serves well for both resharding and reshape) and rename it to collect_output_unshared_sstables().

True. I wonder if it makes sense to keep the sanity check that after reshape all sstables go to local-unshared?

Reply:
It does, as reshape and reshard will always produce unshared sstables. The former produce unshared at current shard, whereas the latter produce unshared at either current or a single remote shard. So the sanity check, which output sstable belongs to a single shard, works for both.